### ISSUE TITLE
NuGetHasSource is case sensitive

### ISSUE THREAD

**Issue by hayhay27** [cake-build/cake#2030]
### What You Are Seeing?

With source configuration

> C:\\> nuget source list
>
> Registered Sources:
> 
>   1.  PrivateFeed [Enabled]
>       https://AnYsOuRcE.org/v3/index.json
> 

Next code returns error:
```
Task("AddSource")
.WithCriteria(() => !NuGetHasSource("https://anysource.org/v3/index.json"))
.Does(() => NuGetAddSource("LocalFeed", "https://anysource.org/v3/index.json"));
```
Error message:
> The source specified has already been added to the list of available package sources. Please provide a unique source.

### What is Expected?
The task should be skipped

### What version of Cake are you using?
0.23

### Are you running on a 32 or 64 bit system?
64

### What environment are you running on?  Windows? Linux? Mac?
Windows

### Are you running on a CI Server?  If so, which one?
Teamcity. Actually doesn't matter

### How Did You Get This To Happen? (Steps to Reproduce)
Add new source
> nuget source add -Name LocalFeed -Source https://AnySource.org/nuget

Run simple cake script
```
Task("AddSource")
.WithCriteria(() => !NuGetHasSource("https://anysource.org/nuget"))
.Does(() => NuGetAddSource("LocalFeed", "https://anysource.org/nuget"));

RunTarget("AddSource");
```

**Comment by coder2213** [comment]
@augustoproiete 

I am able to recreate this issue.
In ProcessRunner.cs, a console process is getting created and when we run this command, the output is getting displayed on the console even though process is still running. Also exitcode is getting set to 1 due to the error I think.

Process class of System.Diagonstics is used to run process. To fix this issue, I need to achieve below:
1. Do not display the output of command run for this.
2. Exitcode should not be set to 1 for this error to avoid throwing exception.

I don't know how to do these and appreciate your help.
Thanks.

**Comment by coder2213** [comment]
@augustoproiete 

I have fixed this. Not sure on appropriate behavior for this.
As per below mentioned code, tasks whose criteria is not met will be thrown.
Should we throw for this issue as per existing code or not do anything?

```
Error: One or more errors occurred.
        Could not reach target 'AddSource' since it was skipped due to a criteria.
```




  ```
private static bool ShouldTaskExecute(ICakeContext context, CakeTask task, CakeTaskCriteria criteria, bool isTarget)
        {
            if (!criteria.Predicate(context))
            {
                if (isTarget)
                {
                    // It's not OK to skip the target task.
                    // See issue #106 (https://github.com/cake-build/cake/issues/106)
                    const string format = "Could not reach target '{0}' since it was skipped due to a criteria.";
                    var message = string.Format(CultureInfo.InvariantCulture, format, task.Name);
                    throw new CakeException(message);
                }

                return false;
            }

            return true;
        }
```

**Comment by coder2213** [comment]
@augustoproiete 
Thanks for assigning this to me.

Please confirm on appropriate behavior.
Fix is very simple, I can give you PR immediately.

**Comment by augustoproiete** [comment]
@coder2213 The behavior of `WithCriteria` is correct (or at least out of scope for this issue). The issue described is that if we call `NuGetAddSource("Bla", "https://bla")` and then check `NuGetHasSource("https://BLA")`, it returns `false`, because the comparison is case sensitive (but should return `true` instead given the feed exists already and differs by casing only).

Example repro with Cake:

```cakescript
#tool "nuget:?package=NuGet.CommandLine&version=5.9.1"

Task("AddSource")
    .Does(() =>
    {
        if (!NuGetHasSource("https://anysource.org/nuget"))
        {
            NuGetAddSource("LocalFeed", "https://anysource.org/nuget");
        }

        if (!NuGetHasSource("https://AnYsOuRcE.org/nuget"))
        {
            Warning("If you're seeing this, it means NuGetHasSource is case sensitive");
        }
    });

RunTarget("AddSource");
```

Example repro with NuGet command-line:

```powershell
> NuGet.exe sources Add -Name "BrandNewFeed1" -Source "https://bla" -NonInteractive
# Package source with Name: BrandNewFeed1 added successfully.

> NuGet.exe sources Add -Name "BrandNewFeed2" -Source "https://BLA" -NonInteractive
# The source specified has already been added to the list of available package sources. Provide a unique source.
```

**Comment by augustoproiete** [comment]
Closed by #3404
